variables.tf

```diff
@@ -369,95 +369,109 @@ variable "custom_security_group_ids" {
-  }
+variable "logs_agent_tolerations" {
+  description = "List of tolerations to apply to Logs agent. The default value means a pod will run on every node."
+  type = list(object({
+    key               = optional(string)
+    operator          = optional(string)
+    value             = optional(string)
+    effect            = optional(string)
+    tolerationSeconds = optional(number)
+  }))
+  default = [{
+    operator = "Exists"
+  }]
 }
 
-variable "log_analysis_agent_tags" {
+variable "logs_agent_additional_log_source_paths" {
   type        = list(string)
-  description = "List of tags to associate with the log analysis agents"
+  description = "The list of additional log sources. By default, the Logs agent collects logs from a single source at `/var/log/containers/*.log`."
   default     = []
+  nullable    = false
 }
 
-variable "log_analysis_agent_custom_line_inclusion" {
-  description = "Log Analysis agent custom configuration for line inclusion setting LOGDNA_K8S_METADATA_LINE_INCLUSION. See https://github.com/logdna/logdna-agent-v2/blob/master/docs/KUBERNETES.md#configuration-for-kubernetes-metadata-filtering for more info."
-  type        = string
-  default     = null
+variable "logs_agent_exclude_log_source_paths" {
+  type        = list(string)
+  description = "The list of log sources to exclude. Specify the paths that the Logs agent ignores."
+  default     = []
+  nullable    = false
 }
 
-variable "log_analysis_agent_custom_line_exclusion" {
-  description = "Log Analysis agent custom configuration for line exclusion setting LOGDNA_K8S_METADATA_LINE_EXCLUSION. See https://github.com/logdna/logdna-agent-v2/blob/master/docs/KUBERNETES.md#configuration-for-kubernetes-metadata-filtering for more info."
-  type        = string
-  default     = null
+variable "logs_agent_selected_log_source_paths" {
+  type        = list(string)
+  description = "The list of specific log sources paths. Logs will only be collected from the specified log source paths. If no paths are specified, it will send logs from `/var/log/containers`."
+  default     = []
+  nullable    = false
 }
 
-variable "log_analysis_agent_name" {
-  description = "Log Analysis agent name. Used for naming all kubernetes and helm resources on the cluster."
-  type        = string
-  default     = "logdna-agent"
+variable "logs_agent_log_source_namespaces" {
+  type        = list(string)
+  description = "The list of namespaces from which logs should be forwarded by agent. If namespaces are not listed, logs from all namespaces will be sent."
+  default     = []
   nullable    = false
 }
 
-variable "log_analysis_agent_namespace" {
+variable "logs_agent_iam_mode" {
   type        = string
-  description = "Namespace where to deploy the Log Analysis agent. Default value is 'ibm-observe'"
-  default     = "ibm-observe"
-  nullable    = false
+  default     = "TrustedProfile"
+  description = "IAM authentication mode: `TrustedProfile` or `IAMAPIKey`."
```

Comment:
This should mention that if `TrustedProfile` is selected, the module will create one. Mention the exact scope its created with too.
We probably should also support using existing trusted profile too (but I guess can be added later)